redis_collections/sets.py

```diff
@@ -312,62 +39,146 @@ def __init__(self, *args, **kwargs):
             make your own implementation by subclassing and overriding
             internal method :func:`_create_key`.
         """
+        self._pickle = self._pickle_2 if six.PY2 else self._pickle_3
```

Comment:
I realize this makes subclassing and overriding `_pickle` and `_unpickle` difficult; I'll fix that in my next PR.